docs/WalletGuide.md

```diff
@@ -699,3 +707,224 @@ suspend fun main() {
+val account = "<account address>".toPublicKeyPair()
+
+suspend fun accountInfo() {
+-->
+
+<!--- SUFFIX 
+}
+-->
+
+You can fetch account info from one or more servers. To do so, first we need to authenticate,
+```kotlin
+  val auth1 = recovery.sep10Auth(first).authenticate(recoveryKey)
+  val auth2 = recovery.sep10Auth(second).authenticate(recoveryKey)
+```
+This implementation uses SEP-10 as authentication method.
+
+Next, get account info using auth tokens:
+```kotlin
+  val accountInfo = recovery.getAccountInfo(account, mapOf(first to auth1, second to auth2))
+
+  println("Recoverable info: $accountInfo")
+```
+
+The result info will be associated with a respective recovery server key.
+
+> You can get the full code [here](../examples/documentation/src/example-recovery-03.kt).
+
+### Recover account
+
+<!--- INCLUDE
+val recoveryKey = SigningKeyPair.fromSecret("<recovery key>")
+val account = "<account address>".toPublicKeyPair()
+val recoverySigners = listOf("<first signer address>", "<second signer address>")
+
+suspend fun recover() {
+-->
+
+<!--- SUFFIX 
+}
+-->
+
+#### Prepare for recovery
+**Note**: This section is when recovery server doesn't support signing sponsored transaction. If your SEP-30 provider supports it, you can skip to [the next part](#prepare-for-recovery)
+
+With no sponsor it's required to refill account to pay for new key reserves:
+
+```kotlin
+  val refill =
+    wallet
+        .stellar()
+        .transaction(sponsor)
+        .transfer(account.address, XLM, "0.5")
+        .build()
+        .sign(sponsor)
+
+  wallet.stellar().submitTransaction(refill)
+```
+
+#### Recover
+
+First, we need to authenticate with recovery servers,
+```kotlin
+  val auth1 = recovery.sep10Auth(first).authenticate(recoveryKey)
+  val auth2 = recovery.sep10Auth(second).authenticate(recoveryKey)
+```
+We need to know signers that will be used to sign the transaction. You can get them either on wallet creation step (`recoverySigners` variable), or via fetching account info from recovery serves. 
+
+Next, create a new device key and retrieve a signed key replacement transaction from recovery servers:
+```kotlin
+  val newKey = wallet.stellar().account().createKeyPair()
+
+  val serverAuth = mapOf(
+    first to RecoveryServerSigning(recoverySigners[0], auth1),
+    second to RecoveryServerSigning(recoverySigners[1], auth2)
+  )
+
+  val signedReplaceKeyTransaction =
+    recovery.replaceDeviceKey(
+      account,
+      newKey,
+      serverAuth
+    )
+```
+Calling with function will create a transaction that locks previous device key and replaces it with a new key (having the same weight as an old one). 
+Lost device key is deduced automatically. Signer will be considered a device key, if one of these conditions matches:
+1. It's the only signer that's not in `serverAuth`
+2. All signers in `serverAuth` has the same weight, and potential signer is the only one with a different weight.
```

Comment:
What if the old device key doesn't match the criteria here? Can't we specify the old device public key as a parameter to avoid ambiguity?

Reply:
Yes